change-validation/src/main/scala/com/normation/plugins/changevalidation/WorkflowJdbcRepository.scala

```diff
@@ -63,36 +63,60 @@ trait WoWorkflowRepository {
   def updateState(crId: ChangeRequestId, from: WorkflowNodeId, state: WorkflowNodeId): Box[WorkflowNodeId]
 }
 
-class RoWorkflowJdbcRepository(doobie: Doobie) extends RoWorkflowRepository with Loggable {
+trait RoWorkflowJdbcRepositorySQL {
+  def getAllByStateSQL(state: WorkflowNodeId): Query0[ChangeRequestId] = {
+    sql"select id from workflow where state = $state".query[ChangeRequestId]
+  }
 
-  import doobie._
+  def getStateOfChangeRequestSQL(crId: ChangeRequestId): Query0[WorkflowNodeId] = {
+    sql"select state from workflow where id = $crId".query[WorkflowNodeId]
+  }
 
-  val SELECT_SQL = fr"SELECT id, state FROM Workflow "
+  def getAllChangeRequestsStateSQL: Query0[(ChangeRequestId, WorkflowNodeId)] = {
+    sql"select id, state from workflow".query[(ChangeRequestId, WorkflowNodeId)]
+  }
+}
+
+trait WoWorkflowJdbcRepositorySQL {
+  def createWorkflowSQL(crId: ChangeRequestId, state: WorkflowNodeId): Update0 = {
+    sql"insert into workflow (id, state) values ($crId, $state)".update
+  }
+
+  def updateStateSQL(crId: ChangeRequestId, from: WorkflowNodeId, state: WorkflowNodeId): Update0 = {
+    sql"update workflow set state = $state where id = $crId".update
+  }
+}
+trait WorkflowJdbcRepositorySQL extends RoWorkflowJdbcRepositorySQL with WoWorkflowJdbcRepositorySQL {}
+object WorkflowJdbcRepositorySQL extends WorkflowJdbcRepositorySQL
```

Comment:
Can you move `WorkflowJdbcRepositorySQL` into test since it's not use here?
Plus, I think the trait `WorkflowJdbcRepositorySQL` isn't use, you could just have in the test file `object WorkflowJdbcRepositorySQL extends RoWorkflowJdbcRepositorySQL with WoWorkflowJdbcRepositorySQL {}` no ?

Reply:
The solution was as you suggested to remove the unnecessary trait, the `object  WorkflowJdbcRepositorySQL` is used in the file here (to implement the repos) as well as in query type-checking tests, so there is only a single object that contains all the SQL now : https://github.com/Normation/rudder-plugins/pull/618/commits/3fbb60377fcd6ac1c278b9f60319f44797af6e1f